From 9fe8694282b988d946b62b11a3b771ea2ec30d25 Mon Sep 17 00:00:00 2001
From: Jukka Rissanen <jukka.rissanen@linux.intel.com>
Date: Wed, 18 Apr 2012 17:57:21 +0300
Subject: [PATCH 2/5] pacrunner: Memory leak in dbus container

Valgrind reported memory leak here. The dbus message container
must be not left open because it has allocated resources so we
close the container before returning.
---
 plugins/pacrunner.c |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/plugins/pacrunner.c b/plugins/pacrunner.c
index 0a9fa55..ba80fcb 100644
--- a/plugins/pacrunner.c
+++ b/plugins/pacrunner.c
@@ -121,6 +121,7 @@ static void create_proxy_configuration(void)
 
 	switch(connman_service_get_proxy_method(default_service)) {
 	case CONNMAN_SERVICE_PROXY_METHOD_UNKNOWN:
+		connman_dbus_dict_close(&iter, &dict);
 		goto done;
 	case CONNMAN_SERVICE_PROXY_METHOD_DIRECT:
 		method= "direct";
@@ -129,8 +130,10 @@ static void create_proxy_configuration(void)
 		method = "manual";
 
 		str_list = connman_service_get_proxy_servers(default_service);
-		if (str_list == NULL)
+		if (str_list == NULL) {
+			connman_dbus_dict_close(&iter, &dict);
 			goto done;
+		}
 
 		connman_dbus_dict_append_array(&dict, "Servers",
 					DBUS_TYPE_STRING, append_string_list,
@@ -154,8 +157,10 @@ static void create_proxy_configuration(void)
 		if (str == NULL) {
 			str = connman_service_get_proxy_autoconfig(
 							default_service);
-			if (str == NULL)
+			if (str == NULL) {
+				connman_dbus_dict_close(&iter, &dict);
 				goto done;
+			}
 		}
 
 		connman_dbus_dict_append_basic(&dict, "URL",
-- 
1.7.10

